File: src/services/categoryService.js
import api from "./api";

const getAllCategories = () => api.get("/categories");
const getCategoryById = (id) => api.get(`/categories/${id}`);
const createCategory = (category) => api.post("/categories", category);
const updateCategory = (id, category) => api.put(`/categories/${id}`, category);
const deleteCategory = (id) => api.delete(`/categories/${id}`);

export default {
  getAllCategories,
  getCategoryById,
  createCategory,
  updateCategory,
  deleteCategory,
};

File: inventory-backend/src/main/java/com/example/inventory_backend/service/CategoryService.java
package com.example.inventory_backend.service;

import com.example.inventory_backend.model.Category;
import com.example.inventory_backend.repository.CategoryRepository;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class CategoryService {

    private final CategoryRepository categoryRepository;

    public CategoryService(CategoryRepository categoryRepository) {
        this.categoryRepository = categoryRepository;
    }

    public List<Category> getAllCategories() {
        return categoryRepository.findAll();
    }

    public Optional<Category> getCategoryById(Long id) {
        return categoryRepository.findById(id);
    }

    public Category createCategory(Category category) {
        return categoryRepository.save(category);
    }

    public Category updateCategory(Long id, Category categoryDetails) {
        return categoryRepository.findById(id).map(category -> {
            category.setName(categoryDetails.getName());
            category.setDescription(categoryDetails.getDescription());
            return categoryRepository.save(category);
        }).orElseThrow(() -> new RuntimeException("Category not found with id " + id));
    }

    public void deleteCategory(Long id) {
        categoryRepository.deleteById(id);
    }
}
